common/utils.go

```diff
@@ -46,13 +47,17 @@ func TransactionLogging(reducedLoggingResponseCodes []int, logger kitlog.Logger)
 	errorLogger := logging.Error(logger)
 	return func(ctx context.Context, code int, r *http.Request) {
 		tid, _ := ctx.Value(ContextKeyRequestTID).(string)
-		transactionInfoLogger, ok := ctx.Value(ContextKeyTransactionInfoLogger).(kitlog.Logger)
-		traceId, spanId := candlelight.ExtractTraceInformation(r.Context())
-
-
+		transactionInfoLogger, transactionLoggerOk := ctx.Value(ContextKeyTransactionInfoLogger).(kitlog.Logger)
+		traceId, spanId, traceOk := candlelight.ExtractTraceInformation(r.Context())
```

Comment:
This might be less noisy if `candlelight.ExtractTraceInformation` actually did the appending.  More like:

```
traceLogKVs := candlelight.AppendTraceInformation(r.Context(), nil)
```

Then, all the logic based around `traceOk` can just go inside the `candlelight` function.

This reduces all the extra code a client has to do.  If a client needs to know if tracing was appended, then returning a `[]interface{}, bool` can address that.

Reply:
I should point out that the signature would be something like:

```
func AppendTraceInformation(ctx context.Context, kvs []interface{}) []interface{} /* maybe add a bool */ {
    kvs = append( /* stuff */ )
    return kvs
}